### PR TITLE
chore: speed up builds

### PR DESCRIPTION
**Pull request by craigbilner**
Move linting and tests into custom script and split across four separate Travis jobs which cover:

* android
* ios
* web
* rest (builds the storybook, cli tests, linting, platform agnostic tests)

Even if the tests don't go faster they shouldn't time out on a single job and still allow us to run all tests

n.b. also fixed some dep issues and test script breakage which seems to have made testing gestures impossible with fake timers....

### CODE REVIEW

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/dcf8fa8cce88f871fd6d6f2b6ccd10935e9957a6/dextrose-test/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/dd4dcbb2a443bc65624ca3932a43e1fed68a835f/dextrose-test/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/a8d03eec9b0cd7a5f6265102a95ff8655056c4cc/dextrose-test/index.html

File: packages/gestures/__tests__/gestures.native.test.js [hunk 1]
```diff
@@ -179,7 +179,7 @@ export default () => {
       expect(component.find("AnimatedComponent")).toMatchSnapshot();
     });
 
-    it("animate back to the start after release", async () => {
+    xit("animate back to the start after release", async () => {
```
Comment (L0wry) [inline packages/gestures/__tests__/gestures.native.test.js]: xit

Comment (craigbilner) [reply packages/gestures/__tests__/gestures.native.test.js]: I don't want to block this PR which has more value than this test which looks like it's going to be a pain to get working

Comment (L0wry) [reply packages/gestures/__tests__/gestures.native.test.js]: sure thing bossman 👍

**Comment by jimhunty** [comment]
Cant we have a test script for each package that does them all?

File: packages/gestures/__tests__/gestures.native.test.js (hunk 1 above)
Comment (craigbilner) [reply packages/gestures/__tests__/gestures.native.test.js]: I will fix it shortly after though :-)

Comment (rjanjua) [reply packages/gestures/__tests__/gestures.native.test.js]: *2 years later*

**Pull request merged**